src/CameraControls.ts

```diff
@@ -1019,25 +1019,11 @@ export class CameraControls extends EventDispatcher {
 
 				}
 
-			} else {
-
-				if ( ! this._lockedPointer && ( event.buttons & MOUSE_BUTTON.LEFT ) === MOUSE_BUTTON.LEFT ) {
-
-					this._state = this._state | this.mouseButtons.left;
-
-				}
-
-				if ( ( event.buttons & MOUSE_BUTTON.MIDDLE ) === MOUSE_BUTTON.MIDDLE ) {
-
-					this._state = this._state | this.mouseButtons.middle;
+			} else if ( ! this._lockedPointer && ( event.buttons & MOUSE_BUTTON.LEFT ) === MOUSE_BUTTON.LEFT ||
+				( event.buttons & MOUSE_BUTTON.MIDDLE ) === MOUSE_BUTTON.MIDDLE ||
+				( event.buttons & MOUSE_BUTTON.RIGHT ) === MOUSE_BUTTON.RIGHT ) {
```

Comment:
I do not use `MOUSE_BUTTON.ALL` here because having `! this._lockedPointer &&` changes the logic.